homeassistant/components/climate/generic_thermostat.py

```diff
@@ -287,23 +367,27 @@ def _async_sensor_changed(self, entity_id, old_state, new_state):
             return
 
         self._async_update_temp(new_state)
-        self._async_control_heating()
+        self._async_control_thermostat()
         yield from self.async_update_ha_state()
 
     @callback
     def _async_switch_changed(self, entity_id, old_state, new_state):
-        """Handle heater switch state changes."""
+        """Handle switch state changes."""
         if new_state is None:
             return
         self.async_schedule_update_ha_state()
 
     @callback
     def _async_keep_alive(self, time):
```

Comment:
This is a weird method. Shouldn't this just be keeping track of our heat or cool switches and if they turn off/on outside of our control, we change them back? ( I guess this would be  for a future PR)

Reply:
This was functionality that was implemented previously. From what I understood from the discussions, some devices (like some AC units) need a keep alive sent from the remote to keep them from going into standby.